crates/services/txpool_v2/src/pool_worker.rs

```diff
@@ -377,8 +395,57 @@ where
                         tracing::error!("Failed to send removed notification: {}", e);
                     }
                 }
+                let updated_txs = self.pending_pool.new_known_tx(tx);
+
+                for (tx, source) in updated_txs.resolved_txs {
+                    self.insert(tx, source);
```

Comment:
While I like the fact that we will insert transactions without delay, this can be very long recursion leading to stack overflow. So I will suggest to queue transaction via the channel=)

Reply:
Done.